05-Mining-Protocol.md

```diff
@@ -1,106 +1,195 @@
 
-The protocol dedicates all directly modifiable bits (`version`, `nonce`, and `nTime`) from the block header to one mining channel.
-This is the smallest assignable unit of search space by the protocol.
-The client which opened the particular channel owns the whole assigned space and can split it further if necessary (e.g. for multiple hashing boards and for individual chips etc.).
+Mining servers MUST assign a unique subset of the search space to each mining device, otherwise the mining devices will waste energy with overlapping search.
 
-### 5.1.2 Extended Channels
+This protocol explicitly expects that upstream server software is able to manage the size of the hashing space correctly for its clients and can provide new and unique Jobs quickly enough, based on the hashpower of each client.
 
-Extended channels are intended to be used by proxies.
-Upstream servers which accept connections and provide work MUST support extended channels.
-Clients, on the other hand, do not have to support extended channels, as they MAY be implemented more simply with only standard channels at the end-device level.
-Thus, upstream servers providing work MUST also support standard channels.
+The protocol defines two main types of Jobs: **Standard Jobs** and **Extended Jobs**.
 
-The size of search space for an extended channel is `2^(NONCE_BITS+VERSION_ROLLING_BITS+extranonce_size*8)` per `nTime` value.
+This separation vastly simplifies the protocol implementation for clients that don’t support Extended Jobs, as they only need to implement the subset of protocol messages related to Standard Jobs (see Mining Protocol Messages for details).
 
-### 5.1.3 Group Channels
+Additionally, a Job (either Standard or Extended) also could be potentially labeled as a **Future Job** and/or **Custom Job**.
 
-Standard channels opened within one particular connection can be grouped together to be addressable by a common communication group channel.
+### 5.1.1 Standard Jobs
+
+Standard Jobs are restricted to fixed Merkle Roots. We call this header-only mining (HOM).
+
+The size of the search space for one Standard Job, given a fixed `nTime` field, is `2^(NONCE_BITS + BIP320_VERSION_ROLLING_BITS) = ~280Th`, where `NONCE_BITS = 32` and `BIP320_VERSION_ROLLING_BITS = 16`.
+This is a guaranteed space before `nTime` rolling.
+
+Standard Jobs are distributed via the [`NewMiningJob`](#5415-newminingjob-server---client) message, which can ONLY be sent via [Standard Channels](#531-standard-channels).
+
+All SV2 mining devices with hashrate under 280 TH/s should be restricted to HOM via Standard Jobs. This is a big difference from legacy SV1 mining devices, where rolling extranonces is a common feature. Hashing space optimization should be achieved upstream, where efficient telemetry is achieved by keeping track of multiple Standard Jobs via Group Channels.
```

Comment:
I disagree with this paragraph. What does it even mean? Restricted by whom? Who enforces the restriction?

I'd prefer saying something like
```
Note that mining devices with higher hash rate than 280 TH/s will outrun the true time in nTime field.
It's not necessarily a problem, as long as the difference is not enormous and as long the device keeps receiving mining jobs in small enough intervals.

Client application is responsible for operating the session within sane limits. Alternatively it may request an extended channel.
```

Reply:
Also I'm not quite sure what this means:

```
Hashing space optimization should be achieved upstream, where efficient telemetry is achieved by keeping track of multiple Standard Jobs via Group Channels.
```

* What is hashing space optimization?
* What does efficient telemetry mean? And what does it have to do with standard jobs?